content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,675 @@
+immutable-configmap-volume-78b6fbff95-7vcj4   1/1     Running   0          62s
+immutable-configmap-volume-78b6fbff95-vdslm   1/1     Running   0          62s
+```
+
+The Pod's container refers to the data defined in the ConfigMap and uses it to print a report to stdout. You can check this report by viewing the logs for one of the Pods in that Deployment:
+```shell
+# Pick one Pod that belongs to the Deployment, and view its logs
+kubectl logs deployments/immutable-configmap-volume
+```
+
+You should see an output similar to:
+```
+Found 3 pods, using pod/immutable-configmap-volume-78b6fbff95-5gsfh
+Wed Mar 20 03:52:34 UTC 2024 The name of the company is ACME, Inc.
+Wed Mar 20 03:52:44 UTC 2024 The name of the company is ACME, Inc.
+Wed Mar 20 03:52:54 UTC 2024 The name of the company is ACME, Inc.
+```
+
+{{< note >}}
+Once a ConfigMap is marked as immutable, it is not possible to revert this change nor to mutate the contents of the data or the binaryData field.  
+In order to modify the behavior of the Pods that use this configuration, you will change the name of
+the ConfigMap, and edit the Deployment to define a slightly different pod template, referencing the
+new ConfigMap.
+{{< /note >}}
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap company-name-20150801
+```
+
+In the editor that appears, perform the following actions:
+- change the name of the ConfigMap from `company-name-20150801` to `company-name-20240312`
+- change the value of key `company_name` from `ACME, Inc.` to `Fiktivesunternehmen GmbH`.
+
+Here's an example of how that manifest could look after you edit it:
+
+```yaml
+apiVersion: v1
+data:
+  company_name: "Fiktivesunternehmen GmbH" # new fictional company name
+immutable: true
+kind: ConfigMap
+# Apart from the name field, you can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"v1","data":{"company_name":"ACME, Inc."},"immutable":true,"kind":"ConfigMap","metadata":{"annotations":{},"name":"company-name-20150801","namespace":"default"}}
+  creationTimestamp: "2024-03-20T03:51:47Z"
+  name: company-name-20240312 # new ConfigMap name
+  namespace: default
+  resourceVersion: "854"
+  uid: bc6d7c83-1b01-43d2-acb9-c4e1a4e4a8e8
+```
+
+Attempt to save the changes. This action is refused as the name of the ConfigMap was changed. Instead, a copy of the changes is stored in a temporary manifest as shown below:
+
+```
+A copy of your changes has been stored to "/var/folders/nn/bb6y1j8d69n9lhqppjlrfthm0000gn/T/kubectl-edit-1440698061.yaml"
+error: At least one of apiVersion, kind and name was changed
+```
+
+Create a new ConfigMap by referencing the temporary manifest.
+```shell
+kubectl apply -f /var/folders/nn/bb6y1j8d69n9lhqppjlrfthm0000gn/T/kubectl-edit-1440698061.yaml
+```
+
+You should see an output similar to:
+```
+configmap/company-name-20240312 created
+```
```

Comment:
Should they delete the old ConfigMap?

Reply:
It is not necessary to delete the old ConfigMap.